### PR TITLE
New Package: odin-lang.Odin  version dev-2024-03

### PR DESCRIPTION
**Pull request by og-mrk**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/148123)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:


(Automated response - build 876.)

**Comment by stephengillie** [comment]
Manual Validation ended with:
> PS C:\Users\User\Desktop>  C:\Users\User\AppData\Local\Microsoft\WinGet\Links\odin-windows-amd64-dev-
> 2024-03.exe Program 'odin-windows-amd64-dev-
> 2024-03.exe' failed to run: The specified executable is not a valid application for this OS platform.At line:1 char:2 +  C:\Users\User\AppData\Local\Microsoft\WinGet\Links\odin-windows-amd6 ... +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~. At line:1 char:2 +  C:\Users\User\AppData\Local\Microsoft\WinGet\Links\odin-windows-amd6 ... +  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException     + FullyQualifiedErrorId : NativeCommandFailed  PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\odin-lang.Odin__DefaultSource\odin-windows-amd64-dev-
> 2024-03.exe Program 'odin-windows-amd64-dev-
> 2024-03.exe' failed to run: The specified executable is not a valid application for this OS platform.At line:1 char:1 + C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\odin-lang.Odin_ ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~. At line:1 char:1 + C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\odin-lang.Odin_ ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException     + FullyQualifiedErrorId : NativeCommandFailed 

(Automated response - build 876.)

**Comment by og-mrk** [comment]
My bad @stephengillie
Double-checked the installer manifest file, noticed it's abit odd when compared to a normal `.zip` portable install, patched it in a new force-pushed commit.

> [!NOTE]
> The Validation will probably fail, because after the Installation has finished successfully, running `odin.exe` **without** any arguments will return exit code 1 , which's generally considered a Program Failure for most apps, but in reality in this case, it's just informing the user of this program that you need to provide extra arguments so you can use it properly, **_see screenshot below for context/evidence_**.
> 
> </br>
> 
> image

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Executable C:\Users\validator\AppData\Local\Microsoft\WinGet\Packages\odin-lang.Odin__DefaultSource\odin.exe returned exit code: -1073741515
> Executable C:\Users\validator\AppData\Local\Microsoft\WinGet\Links\odin.exe returned exit code: -1073741515

(Automated response - build 876.)

File: manifests/o/odin-lang/Odin/dev-2024-03/odin-lang.Odin.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Created with YamlCreate.ps1 v2.4.1 $debug=NVS2.CRLF.7-4-1.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: odin-lang.Odin
+PackageVersion: dev-2024-03
+InstallerType: zip
+FileExtensions:
+- odin
+ReleaseDate: 2024-03-04
+Installers:
```
Comment (stephengillie) [inline manifests/o/odin-lang/Odin/dev-2024-03/odin-lang.Odin.installer.yaml:10]:
```suggestion
Dependencies:
  PackageDependencies:
     - PackageIdentifier: Microsoft.VCRedist.2015+.x64
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
image

**Comment by stephengillie** [comment]
Verify:
image

**Review by stephengillie — APPROVED** [review]

**Pull request merged**